qiskit/transpiler/passes/routing/basic_swap.py

```diff
@@ -98,12 +108,79 @@ def run(self, dag):
                     # update current_layout
                     for swap in range(len(path) - 2):
                         current_layout.swap(path[swap], path[swap + 1])
-
-            order = current_layout.reorder_bits(new_dag.qubits)
-            new_dag.compose(subdag, qubits=order)
+            # handle control flow operations
+            for node in subdag.control_flow_ops():
+                updated_ctrl_op, cf_layout = self._transpile_controlflow_op(node.op, current_layout)
+                node.op = updated_ctrl_op
+                cf_layer = True
+            if cf_layer:
+                new_dag.compose(subdag)
+                current_layout = cf_layout
+            else:
+                order = current_layout.reorder_bits(new_dag.qubits)
+                new_dag.compose(subdag, qubits=order)
+        self.property_set["final_layout"] = current_layout
 
         return new_dag
 
+    def _transpile_controlflow_op(self, cf_op, current_layout):
+        """default function"""
+        if isinstance(cf_op, IfElseOp):
+            return self._transpile_controlflow_multiblock(cf_op, current_layout)
+        elif isinstance(cf_op, (ForLoopOp, WhileLoopOp)):
+            return self._transpile_controlflow_looping(cf_op, current_layout)
+        return cf_op, current_layout
+
+    def _transpile_controlflow_multiblock(self, cf_op, current_layout):
+        # pylint: disable=cyclic-import
+        from qiskit.transpiler.passes.routing.layout_transformation import LayoutTransformation
+        from qiskit.converters import dag_to_circuit, circuit_to_dag
+
+        block_circuits = []  # control flow circuit blocks
+        block_dags = []  # control flow dag blocks
+        block_layouts = []  # control flow layouts
+
+        for i, block in enumerate(cf_op.blocks):
```

Comment:
```suggestion
        for block in cf_op.blocks:
```
The `enumerate` doesn't seem to be used.